src/com/etendoerp/etendorx/config/ConnectorPropertiesInjector.java

```diff
@@ -0,0 +1,78 @@
+package com.etendoerp.etendorx.config;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.codehaus.jettison.json.JSONException;
+import org.codehaus.jettison.json.JSONObject;
+import org.openbravo.base.exception.OBException;
+import org.openbravo.dal.core.OBContext;
+import org.openbravo.dal.service.OBDal;
+import org.openbravo.erpCommon.utility.Utility;
+import org.openbravo.model.ad.access.User;
+import org.openbravo.service.db.DalConnectionProvider;
+
+import com.etendoerp.etendorx.data.ETRXoAuthProvider;
+import com.etendoerp.etendorx.data.InstanceConnector;
+import com.etendoerp.etendorx.utils.OAuthProviderConfigInjector;
+import com.smf.securewebservices.utils.SecureWebServicesUtils;
+
+/**
+ * This class is responsible for injecting configuration properties.
+ */
+public class ConnectorPropertiesInjector implements OAuthProviderConfigInjector {
+  private static final Logger log = LogManager.getLogger();
+  private static final String WORKER_SERVICE = "worker";
+
+  /**
+   * Injects configuration data into the provided JSONObject.
+   * The configuration data is sourced from the sourceJSON parameter.
+   *
+   * @param sourceJSON the JSONObject that contains the configuration data to be injected
+   * @throws JSONException if an error occurs while injecting the configuration data
+   */
+  @Override
+  public void injectConfig(JSONObject sourceJSON) throws JSONException {
+    try {
+      if (StringUtils.equals(WORKER_SERVICE, sourceJSON.getString("name"))) {
+        InstanceConnector instanceConnector = (InstanceConnector) OBDal.getInstance().createCriteria(
+                InstanceConnector.class)
+            .setMaxResults(1)
+            .uniqueResult();
+        if (instanceConnector == null) {
+          throw new OBException("No InstanceConnector found.");
+        }
+        User tokenUser = instanceConnector.getUserForToken();
+        String token = SecureWebServicesUtils.generateToken(tokenUser);
+        JSONObject propertySources = sourceJSON.getJSONArray("propertySources")
+            .getJSONObject(0)
+            .getJSONObject("source");
+        propertySources.put("connector.instance", instanceConnector.getId());
+        propertySources.put("token", token);
+        propertySources.put("classic.token", token);
+      }
+    } catch (NullPointerException npe) {
```

Comment:
Instead of capture this exception you must check the code to be null safe.

Reply:
Fixed!